### DISCUSSION TITLE
Combine catalog name, schema name and table name together!

### DISCUSSION THREAD

**Discussion by waynexia** [GreptimeTeam/greptimedb#580] [General]
As @MichaelScofield brings up in #559, there are lots of variants in the codebase to express the combination of catalog name, schema name and table name. Looks awful, and we have to unify them. But apparently, an anonymous triple-tuple is not good, the unified one should be a named struct. Then comes the question, what's the name 😋

@MichaelScofield, @iwinux and me proposed several options as below. This is very interesting and makes me open a vote for it (and by the way have a try on this new feature of GH). Please make a vote if you've noticed this discussion, and of course it's ok to propose other options to make it more chaos 😈

**Comment by evenyag** [comment]
There is a terminology `fully qualified table name` for this. Does this struct also need to represent cases like `schema.table`, `table`? If it does, then `TableName` may be more straightforward.

For the fully qualified version, I would vote for `TableFullName` as `TablePath` sounds like a storage path. Another candidate might be `FullyQualifiedName`, though it looks a bit complicated.

**Comment by v0y4g3r** [comment]
We've already had a `TableName` define in `meta_client::rpc`. Please don't forget to replace it with whatever the vote result will be. 

I'd prefer `TableName`/`TableFullName` TBH.

https://github.com/GreptimeTeam/greptimedb/blob/7ba512980a89b76f333af34fa55099f2ddab4d13/src/meta-client/src/rpc.rs#L101-L105
